### PR TITLE
refactored core-tests into separate sub-project 

### PR DESCRIPTION
**Pull request by MansurAshraf**
Created a separate sub-project 'summingbird-core-test'

### CODE REVIEW

File: summingbird-core-test/src/main/scala/com/twitter/summingbird/MemoryArbitraries.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import com.twitter.summingbird.{ KeyedProducer, Producer }
```
Comment (ianoc) [inline summingbird-core-test/src/main/scala/com/twitter/summingbird/MemoryArbitraries.scala]: This is missing new line at end of file, package statement at the top and the apache header

File: summingbird-core-test/src/test/scala/com/twitter/summingbird/memory/MemoryLaws.scala [hunk 2]
```diff
@@ -87,7 +67,7 @@ object MemoryLaws extends Specification {
    * service functions.
    */
   def leftJoinLaw[T: Manifest: Arbitrary, K: Arbitrary, U: Arbitrary, JoinedU: Arbitrary, V: Monoid: Arbitrary: Equiv] = {
-    import MemoryArbitraries._
+    import com.twitter.summingbird.memory.MemoryArbitraries._
```
Comment (ianoc) [inline summingbird-core-test/src/test/scala/com/twitter/summingbird/memory/MemoryLaws.scala]: The package location of these hasn't changed, so the imports shouldn't need to either?

**Comment by johnynek** [comment]
man it would be cool if our tests worked.

**Comment by sritchie** [comment]
@johnynek Come to me with those concerns - Q2 is already fully booked, but I'll take it into consideration for the Q3 roadmap.

**Comment by johnynek** [comment]
@sritchie too real, man.

**Comment by johnynek** [comment]
@MansurAshraf can we close this one now?

**Comment by MansurAshraf** [comment]
@johnynek no I need this merged for internal refactoring we are doing.

**Comment by MansurAshraf** [comment]
To give a little context I need `TestGraphs` published as an artifact as multiple downstream projects are dependent on it.

**Comment by johnynek** [comment]
Okay, yes. Got it. Let me review.

**Pull request merged**